src/main/java/org/mifos/connector/channel/interceptor/IdInterceptor.java

```diff
@@ -0,0 +1,96 @@
+package org.mifos.connector.channel.interceptor;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.mifos.connector.channel.camel.routes.ChannelRouteBuilder;
+import org.mifos.connector.common.channel.dto.PhErrorDTO;
+import org.mifos.connector.common.exception.PaymentHubError;
+import org.mifos.connector.common.exception.PaymentHubErrorCategory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.mifos.connector.channel.camel.config.CamelProperties.CLIENTCORRELATIONID;
+
+@Component
+@Primary
+public class IdInterceptor  implements HandlerInterceptor {
+
+    @Autowired
+    ChannelRouteBuilder channelRouteBuilder;
+
+    @Value("${redis.idempotency.enabled}")
+    Boolean redisIdempotencyEnabled;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+
+        PhErrorDTO errorDTO = null;
+        String correlationId = request.getHeader(CLIENTCORRELATIONID);
+        logger.debug("X-CorrelationID value is {}", correlationId);
+        if (redisIdempotencyEnabled) {
+            if (Boolean.TRUE.equals(channelRouteBuilder.redisTemplate.hasKey(correlationId))) {
```

Comment:
need empty string check before going to redis

Reply:
also as an else condition here we can add it to Redis. . is there a chance that we use the same correlation id for multiple requests. one for the actual transactions and other for getting the status? if so , please include the  API name (endpoint) as well wile inserting to redis